src/node/genesis_gen.h

```diff
@@ -59,11 +59,6 @@ namespace ccf
       }
     }
 
-    auto finalize()
-    {
-      return tx.commit();
-    }
-
```

Comment:
This meant `GenesisGen` (and everyone creating one) needed a `CommittableTx`, just so we could have hide this final call. Instead I've kept it so `GenesisGen` sees the (now-reduced) `kv::Tx&`, and the caller is responsible for calling `tx.commit()`. In practice this is fine; all of the callers have their `gen_tx` directly available _and_ already know it as a `kv::CommittableTx`.